template/koa-rest-api/src/testing/server.ts

```diff
@@ -10,19 +10,8 @@ import { createApp } from 'src/framework/server';
  * Create a new SuperTest agent from a Koa application.
  */
 export const agentFromApp = <State, Context>(app: Koa<State, Context>) => {
-  let server: Server;
-  let agent: request.SuperTest<request.Test>;
-
-  const getAgent = () => agent;
-
-  const setup = async () => {
-    await new Promise((resolve) => (server = app.listen(undefined, resolve)));
-    agent = request.agent(server);
-  };
-
-  const teardown = () => new Promise((resolve) => server.close(resolve));
-
-  return Object.assign(getAgent, { setup, teardown });
+  const agent: request.SuperTest<request.Test> = request.agent(app.callback());
+  return () => agent;
```

Comment:
I wonder if we should just return `agent` directly from all these helpers? That way we can just do `agent.do()` rather than `agent().do()`

Reply:
👍